### ISSUE TITLE
Side panel page navigation won't work causing page to freeze

### ISSUE THREAD

**Issue by amizz** [phonon-framework/phonon#170]
I'm having an issue with side panel page navigation. It gives out an error about failed to execute 'removeChild'. The page will freeze when I navigate from a page to the other page and then back to the previous page. Example: Page1 > Page2 > Page1. This will cause Page1 to freeze.

**Error**
> phonon.js:4417 Uncaught NotFoundError: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.closed @ phonon.js:4417

**Code**
index.html
```
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8" />
        <meta name="format-detection" content="telephone=no" />
        <meta name="viewport" content="user-scalable=no, initial-scale=1, maximum-scale=1, minimum-scale=1, width=device-width, height=device-height" />

        <link rel="stylesheet" href="dist/css/phonon.css" />
        <title>Side Panel</title>
    </head>
    <body>

        <div class="side-panel side-panel-left" data-expose-aside="left" data-disable="right" data-page="home, pagetwo, pagethree, pagefour" id="side-panel-example">
            <header class="header-bar">
                <button class="btn pull-right icon icon-close show-for-phone-only" data-side-panel-close="true"></button>
                <div class="pull-left">
                    <h1 class="title">Side Panel</h1>
                </div>
            </header>
            <div class="content">
                <ul class="list">
                    <li><a href="#!home" class="padded-list">Home</a></li>
                    <li><a href="#!pagetwo" class="padded-list">Page Two</a></li>
                    <li><a href="#!pagethree" class="padded-list">Page Three</a></li>
                    <li><a href="#!pagefour" class="padded-list">Page Four</a></li>
                </ul>
            </div>
        </div>

        <home data-page="true">
            <header class="header-bar">
                <button class="btn icon icon-menu pull-left show-for-phone-only" data-side-panel-id="side-panel-example"></button>
                <div class="center">
                    <h1 class="title">Home</h1>
                </div>
            </header>
            <div class="content">
                <h1>Home</h1>
            </div>
        </home>
        <pagetwo data-page="true"></pagetwo>
        <pagethree data-page="true"></pagethree>
        <pagefour data-page="true"></pagefour>
        <script src="dist/js/phonon.js"></script>
        <script src="app.js" charset="utf-8"></script>
    </body>
</html>
```

pagetwo.html/pagethree.html/pagefour.html (same structure with different element name based on page)
```
<pagetwo data-page="true">
    <header class="header-bar">
        <button class="btn icon icon-menu pull-left show-for-phone-only" data-side-panel-id="side-panel-example"></button>
        <div class="center">
            <h1 class="title">Page Two</h1>
        </div>
    </header>
    <div class="content">
        <h1>Page Two</h1>
    </div>
</pagetwo>
```

app.js
```
phonon.options({
    navigator: {
        defaultPage: 'home',
        animatePages: true,
        enableBrowserBackButton: true,
        templateRootDirectory: './page'
    },
    i18n: null
});

var app = phonon.navigator();
app.on({page: 'home', preventClose: false, content: null, readyDelay: 1});
app.on({page: 'pagetwo', preventClose: false, content: 'pagetwo.html', readyDelay: 1});
app.on({page: 'pagethree', preventClose: false, content: 'pagethree.html', readyDelay: 1});
app.on({page: 'pagefour', preventClose: false, content: 'pagefour.html', readyDelay: 1});

app.start();
```

**Comment by John-Henrique** [comment]
This happening with me too. Every when navigate between pages.

**Comment by amizz** [comment]
A quick hack

```
//Small hack to remove the backdrop which cause the page to freeze by removing elements with name 'backdrop'
//Based on answer at http://stackoverflow.com/questions/4777077/removing-elements-by-class-name by vepasto
function removeElementsByClass(className){
    var elements = document.getElementsByClassName(className);
    while(elements.length > 0){
        elements[0].parentNode.removeChild(elements[0]);
    }
}

phonon.options({
    navigator: {
        defaultPage: 'home',
        animatePages: false, //Disabled because of quirky animation when navigating between pages
        enableBrowserBackButton: true,
        templateRootDirectory: './page'
    },
    i18n: null
});

var app = phonon.navigator();
app.on({page: 'home', preventClose: false, content: null}, function(activity) {
    activity.onReady(function() {
        removeElementsByClass('backdrop-panel');
    });
});
app.on({page: 'pagetwo', preventClose: false, content: 'pagetwo.html', readyDelay: 1}, function(activity) {
    activity.onReady(function() {
        removeElementsByClass('backdrop-panel');
    });
});
app.on({page: 'pagethree', preventClose: false, content: 'pagethree.html', readyDelay: 1}, function(activity) {
    activity.onReady(function() {
        removeElementsByClass('backdrop-panel');
    });
});
app.on({page: 'pagefour', preventClose: false, content: 'pagefour.html', readyDelay: 1}, function(activity) {
    activity.onReady(function() {
        removeElementsByClass('backdrop-panel');
    });
});

app.start();
```

**Comment by John-Henrique** [comment]
I try this to make this more simple. Thanks!

```
document.on( 'pageopened', function( evt ){
	removeElementsByClass('backdrop-panel');
});
```

**Comment by Qathom** [comment]
Hi @amizz and @John-Henrique,

Thank you for opening this issue.
I think that I found the issue.
The fix will be available in the **v1.3.4** (in 1 hour).

Please tell me if it fixes your issue!

**Comment by John-Henrique** [comment]
@Qathom version 1.3.4 not fix this error. Quoted in #179, using phonon.js error on line 4439...

**Comment by qathom** [comment]
Error fixed in v1.5.1.
Please feel free to reopen this issue if you are still facing a similar issue.
